src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -154,13 +154,17 @@ void writeRow(List<Object> row) {
     }
   }
 
+  void writeTempRow(List<Object> row) {
+    tempData.add(row);
+  }
+
   @Override
   float addTempRow(
       Map<String, Object> row,
       int curRowIndex,
       Map<String, RowBufferStats> statsMap,
       Set<String> formattedInputColumnNames) {
-    return addRow(row, this::writeRow, statsMap, formattedInputColumnNames);
+    return addRow(row, this::writeTempRow, statsMap, formattedInputColumnNames);
```

Comment:
@sfc-gh-azagrebin no easy way to test this because `AbstractRowBuffer.insertRows` has a finally block, which cleans the temporary buffer.
```
} finally {
      this.tempStatsMap.values().forEach(RowBufferStats::reset);
      clearTempRows();
      this.flushLock.unlock();
    }